packages/broadcast-channel/src/index.ts

```diff
@@ -0,0 +1,133 @@
+import { createSignal, JSX, onCleanup } from "solid-js";
+
+type OnMessageCB = (e: MessageEvent<any>) => void;
+
+type TBroadcastChannelInstance = {
+  onMessageCBList: { id: number; cb: OnMessageCB }[];
+  instanceCount: number;
+  instance: {
+    onMessage: (cb: OnMessageCB, options?: boolean | AddEventListenerOptions) => void;
+    postMessage: (props: any) => void;
+    close: () => void;
+    channelName: string;
+    instance: BroadcastChannel;
+  };
+};
+const map: {
+  [key: string]: TBroadcastChannelInstance;
+} = {};
+
+let id = 0;
+const createId = () => {
+  return id++;
+};
+
+export function makeBroadcastChannel(name: string) {
+  if (process.env.SSR)
+    return {
+      onMessage: () => void 0,
+      postMessage: () => void 0,
+      close: () => void 0,
+      channelName: name,
+      instance: {} as unknown as BroadcastChannel
+    };
+
+  const id = createId();
```

Comment:
I guess a symbol could be used in the same way, without the need to increment a counter.
```suggestion
  const id = Symbol()
```

Reply:
updated with [d14d67c](https://github.com/solidjs-community/solid-primitives/pull/263/commits/d14d67c27f872e22cfe4f99a78e747df31d3e283)